src/main/java/domain/Janggi.java

```diff
@@ -0,0 +1,118 @@
+package domain;
+
+import domain.position.Position;
+import domain.position.Route;
+import java.util.ArrayList;
+import java.util.List;
+import domain.unit.Unit;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class Janggi {
+    private final List<Unit> units = new ArrayList<>();
+    private Team turn;
+
+    public Janggi() {
+        turn = Team.CHO;
+        List<Unit> hanUnits = settingUnits(Team.HAN);
+        List<Unit> choUnits = settingUnits(Team.CHO);
+        this.units.addAll(Stream.concat(hanUnits.stream(), choUnits.stream()).toList());
+    }
```

Comment:
아래와 같이 주생성자는 생성자 파라미터를 통해서 이미 생성된 Units과 turn을 주입받도록해보면 어떨까요? 
이런 초기 장기판을 만들어주는 역할을 다른 객체에게 위임해보면 좋을 것 같아요. 

이렇게 변경하면, 생성하는 기능도 테스트가능하고 Janggi 객체에 대한 테스트도 작성하기 쉬워질 것 같아요. 

```suggestion
    private final List<Unit> units;
    private Team turn;

    public Janggi(Team turn, List<Unit> units) {
        this.turn = turn;
        this.units = new ArrayList<>(units)
    }
```

Reply:
`Janggi`가 초기 세팅에 관련되어 있다는 생각에 빠져서 `List<Unit>`을 주입받을 생각을 못했네요.

로키의 말대로 `List<Unit>`같은 경우는 외부에서 주입을 하는 것이 더 깔끔할 것 같아요. 장기에는 상차림이라는 규칙이 있으니까요! 👍 

하지만 장기 게임에서 초나라가 먼저 공격을 시작하는 것이 규칙이라면 생성자에서 `turn = Team.CHO;`와 같이 정해주는 것에 대해서는 어떻게 생각하시나요?